code/modules/mob/living/simple_animal/bot/floorbot.dm

```diff
@@ -368,12 +370,20 @@
 				success = TRUE
 
 		if(success)
-			F = F.make_plating(TRUE) || F
-			if(was_replacing && tilestack)
+			var/area/is_this_maints = get_area(F)
+			if(was_replacing && tilestack)	//turn the tile into plating (if needed), then replace it
+				F = F.make_plating(TRUE) || F
 				tilestack.place_tile(F)
 				if(!tilestack)
 					speak("Requesting refill of custom floor tiles to continue replacing.")
-			else
+			else if(F.broken || F.burnt)	//repair the tile and reset it to be undamaged (rather than replacing it)
+				F.broken = FALSE
+				F.burnt = FALSE
+				F.icon_state = initial(F.icon_state)
```

Comment:
this smells like it might break, you sure inital is the right way to do this?

Reply:
already tested it, `initial(F.icon_state)` checks the tile's icon state when it was first loaded; unfortunately there's no proc for UNbreaking/UNburning a tile, so this was the simplest way to do it

Reply:
but yes, after pretty thorough testing, it didnt break. and if it does in the future, that's most likely an issue of the placed tile itself 🦀